src/lib.rs

```diff
@@ -437,6 +437,28 @@ impl PosixACL {
         }
         Ok(())
     }
+
+    /// Consumes the `PosixACL`, returning the wrapped `acl_t`.
+    /// This can be used directly in FFI calls to the acl library.
+    ///
+    /// To avoid a memory leak, the pointer must either:
+    ///
+    /// - Be converted back to a PosixACL using `PosixACL::from_raw`
+    /// - Have `acl_free` called on it
+    pub fn into_raw(&self) -> acl_sys::acl_t {
+        self.acl
```

Comment:
I don't understand how the "Consumes the `PosixACL`" part works. ISTM that Rust will still call `drop` on the original `PosixACL`, which triggers `acl_free()`, meaning that if you pass the same pointer to `PosixACL::from_raw()` it causes double-free.

Also I believe the `into_raw` API should be marked unsafe still, if it's possible to cause a memory leak. This wasn't the case with the previous proposed `as_raw` API.

I wrote this test case which demonstrates the behaviors:
```rust
#[test]
fn acl_into_from_raw() {
    let acl1 = full_fixture();
    let ptr = acl1.into_raw();

    // Should I be able to call `.entries()` after converting acl1 into raw ptr?
    // The docstring suggest no, but this works.
    assert_eq!(acl1.entries().len(), 8);

    // Causes error:
    // thread 'acl_into_from_raw' panicked at 'Error in acl_free: Invalid argument (os error 22)', src/lib.rs:209:9
    let _acl2 = unsafe { PosixACL::from_raw(ptr) };
}
```

Reply:
Yikes! Thanks for catch, I was in a hurry and pushed the wrong version of this - let me re-push  the fixed one in a bit.

On leaking being unsafe - leaking is typically considered safe in rust programs. For example:
 - the safe stdlib function specifically designed to leak memory (https://doc.rust-lang.org/std/mem/fn.forget.html)
 - `Rc::into_raw` also in the stdlib is not unsafe

The nomicon has a more extended discussion on the topic (https://doc.rust-lang.org/nomicon/leaking.html), and the book references it as well (https://doc.rust-lang.org/book/ch15-06-reference-cycles.html#reference-cycles-can-leak-memory)

> Preventing memory leaks entirely is not one of Rust’s guarantees in the same way that disallowing data races at compile time is, meaning memory leaks are memory safe in Rust.

However - as the maintainer of this crate ultimately the decision is up to you! I'll update the PR with to make `into_raw` be unsafe based on your previous feedback.

Reply:
> On leaking being unsafe - leaking is typically considered safe in rust programs.

Ahh thanks for that background. I didn't understand the implications of what you mentioned about `Rc::into_raw` until now. I agree it makes sense to be consistent with Rust conventions. 

Adding a code comment to one of these resources might be helpful. :)

Reply:
Just to be clear: I agree now that `into_raw` shouldn't be `unsafe`.